File: Backend/ItHappened/ItHappenedDomain/Domain/Tracking.cs
using System;
using System.Collections.Generic;

namespace ItHappenedDomain.Domain
{
  public class Tracking
  {

    public Tracking(string trackingName, string trackingId, 
      DateTimeOffset trackingDate, 
      string scale, 
      string rating, 
      string comment,
      DateTimeOffset dateOfChange, bool isDeleted, 
      List<Event> eventCollection,
      string scaleName,
      string color,
      string geoposition)
    {
      this.scaleName = scaleName;
      this.trackingName = trackingName;
      this.trackingId = trackingId;
      this.trackingDate = trackingDate;
      this.scale = scale;
      this.rating = rating;
      this.comment = comment;
      this.dateOfChange = dateOfChange;
      this.isDeleted = isDeleted;
      this.EventCollection = eventCollection;
      this.color = color ?? "-5658199";
      this.geoposition = geoposition ?? "None";
    }

    public string trackingName { set; get; }
    public string trackingId { set; get; }
    public DateTimeOffset trackingDate { set; get; }
    public string scale { set; get; }
    public string rating { set; get; }
    public string comment { set; get; }
    public DateTimeOffset dateOfChange { set; get; }
    public bool isDeleted { set; get; }
    public string scaleName { set; get; }
    public List<Event> EventCollection {get; set;}
    private string color;

    public string Color
    {
      get => color ?? "-5658199";
      set => color = value ?? "-5658199";
    }

    private string geoposition;
    public string Geoposition {
      get { return geoposition ?? "None"; }
      set { geoposition = value ?? "None"; } }
  }
}

File: Android/ItHappenedAndroid/app/src/main/java/ru/lod_misis/ithappened/Domain/Tracking.java
package ru.lod_misis.ithappened.Domain;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

import java.util.Date;

import io.realm.RealmList;
import io.realm.RealmObject;
import io.realm.annotations.PrimaryKey;
import io.realm.annotations.Required;

/**
 * Created by Ded on 07.06.2018.
 */

public class Tracking extends RealmObject{

    public Tracking(String scaleName, String trackingName,
                    String trackingId, Date trackingDate,
                    String scale, String rating, String comment,String geoposition,
                    RealmList<Event> eventCollection, Date dateOfChange,
                    boolean isDeleted, String color) {
        this.scaleName = scaleName;
        this.trackingName = trackingName;
        this.trackingId = trackingId;
        this.trackingDate = trackingDate;
        this.scale = scale;
        this.rating = rating;
        this.comment = comment;
        this.geoposition=geoposition;
        this.eventCollection = eventCollection;
        this.dateOfChange = dateOfChange;
        this.isDeleted = isDeleted;
    }

    public Tracking(){
    }

    @Expose
    @SerializedName("scaleName")
    public String scaleName = "Шкала";
    @Expose
    public String trackingName;
    @Expose
    public String trackingId;
    @Expose
    @SerializedName("trackingDate")
    public Date trackingDate;
    @Expose
    @SerializedName("scale")
    public String scale;
    @Expose
    @SerializedName("rating")
    public String rating;
    @Expose
    @SerializedName("comment")
    public String comment;
    public String geoposition;
    @Expose
    public RealmList<Event> eventCollection;
    @Expose
    public Date dateOfChange;
    @Expose
    public boolean isDeleted = false;
}
